shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -91,7 +92,7 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
                 currentBlockSize = Math.max(TYPE_PLUS_LENGTH_SIZE, blockSizeSupplier.getAppendBlockSize());
                 bytesLeftInBlock = currentBlockSize;
                 segmentBeingAppendedTo = append.segment;
-                writeMessage(new AppendBlock(session.id), out);
+                writeMessage(new AppendBlock(append.getRequestId(), session.id), out);
```

Comment:
This is only the first of many IDs that may be in the block...
Perhaps we need a different abstraction.